packages/application-components/src/components/custom-views/custom-view-loader/custom-view-loader.spec.tsx

```diff
@@ -0,0 +1,128 @@
+import { reportErrorToSentry } from '@commercetools-frontend/sentry';
+import {
+  screen,
+  renderComponent,
+  waitFor,
+  fireEvent,
+} from '../../../test-utils';
+import CustomViewLoader, { type TCustomView } from './custom-view-loader';
+
+const mockShowNotification = jest.fn();
+
+jest.mock('@commercetools-frontend/sentry');
+jest.mock('@commercetools-frontend/actions-global', () => ({
+  useShowNotification: () => mockShowNotification,
+}));
+
+const TEST_CUSTOM_VIEW: TCustomView = {
+  id: 'd8eafca6-1f89-4a84-b93f-ef94f869abcf',
+  defaultLabel: 'Test Custom View',
+  labelAllLocales: {},
+  url: '/',
+  type: 'custom-panel',
+  typeConfig: {
+    size: 'small',
+  },
+  locators: ['customers.customer-detail.addresses'],
+};
+
+describe('CustomViewLoader', () => {
+  beforeAll(() => {
+    window.MessageChannel = jest.fn(() => ({
+      port1: {
+        onmessage: jest.fn(),
+        postMessage: jest.fn(),
+        close: jest.fn(),
+        onmessageerror: jest.fn(),
+        start: jest.fn(),
+        addEventListener: jest.fn(),
+        removeEventListener: jest.fn(),
+        dispatchEvent: jest.fn(),
+      },
+      port2: {
+        onmessage: jest.fn(),
+        postMessage: jest.fn(),
+        close: jest.fn(),
+        onmessageerror: jest.fn(),
+        start: jest.fn(),
+        addEventListener: jest.fn(),
+        removeEventListener: jest.fn(),
+        dispatchEvent: jest.fn(),
+      },
+    }));
+  });
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should render a custom view', async () => {
+    renderComponent(
+      <CustomViewLoader customView={TEST_CUSTOM_VIEW} onClose={jest.fn()} />
+    );
+
+    const iFrame = screen.getByTitle(
+      `Custom View: ${TEST_CUSTOM_VIEW.defaultLabel}`
+    ) as HTMLIFrameElement;
+    expect(iFrame.getAttribute('id')).toBe(
+      `custom-view-${TEST_CUSTOM_VIEW.id}`
+    );
+    expect(iFrame.getAttribute('src')).toBe(TEST_CUSTOM_VIEW.url);
```

Comment:
I wanted to also check the initial `postMessage` call but I don't find a way to replicate in the test a successful iFrame load.
If anyone has any idea, please share it 🙏